extensions/spaces/spaces.lua

```diff
@@ -1,9 +1,874 @@
+    local targetChild = table.remove(mcSpacesChildren)
+    while targetChild do
+        if targetChild.AXIdentifier == targetIdentifier then break end
+        targetChild = table.remove(mcSpacesChildren)
+    end
+    if not targetChild then
+        return nil, string.format("unable to find target %s for display", targetIdentifier)
+    end
+    return targetChild
+end
+
+-- Public interface ------------------------------------------------------
+
+--- hs.spaces.data_missionControlAXUIElementData(callback) -> None
+--- Function
+--- Generate a table containing the results of `hs.axuielement.buildTree` on the Mission Control Accessibility group of the Dock.
+---
+--- Parameters:
+---  * `callback` - a callback function that should expect a table as the results. The table will be formatted as described in the documentation for `hs.axuielement.buildTree`.
+---
+--- Returns:
+---  * None
+---
+--- Notes:
+---  * Like [hs.spaces.data_managedDisplaySpaces](#data_managedDisplaySpaces), this function is not required for general usage of this module; rather it is provided for those who wish to examine the internal data that makes this module possible more closely to see if there might be other information or functionality that they would like to explore.
+---
+---  * Getting Accessibility elements for Mission Control is somewhat tricky -- they only exist when the Mission Control display is visible, which is the exact time that you can't examine them. What this function does is trigger Mission Control and then builds a tree of the elements, capturing all of the properties and property values while the elements are valid, closes Mission Control, and then returns the results in a table by invoking the provided callback function.
+---    * Note that the `hs.axuielement` objects within the table returned will be invalid by the time you can examine them -- this is why the attributes and values will also be contained in the resulting tree.
+---    * Example usage: `hs.spaces.data_missionControlAXUIElementData(function(results) hs.console.clearConsole() ; print(hs.inspect(results)) end)`
+module.data_missionControlAXUIElementData = function(callback)
+    assert(
+        type(callback) == "nil" or type(callback) == "function" or (getmetatable(callback) or {}).__call,
+        "callback must be nil or a function"
+    )
+
+    openMissionControl()
+    local missionControlGroup, initialTime = nil, os.time()
+    while not missionControlGroup and (os.time() - initialTime) < 2 do
+        missionControlGroup = getMissionControlGroup()
+    end
+    if not missionControlGroup then
+        return nil, "unable to get Mission Control data from the Dock"
+    end
+
+    -- delay to make sure Mission Control has stabilized
+    local time = timer.secondsSinceEpoch()
+    while timer.secondsSinceEpoch() - time < module.MCwaitTime do
+        -- twiddle thumbs, calculate more digits of pi, whatever floats your boat...
+    end
+
+    local tree
```

Comment:
I forget, how do I tell stickler to ignore this? The variable is made local and then defined in the next line because it's referenced as an up-value to prevent collection, but it's not referenced again *outside* of the callback function, which is what I expect is ticking off stickler...

Reply:
You can use:

```lua
-- luacheck:ignore
```

You can also `push` and `pop`:

https://luacheck.readthedocs.io/en/stable/inline.html